Do not generate long log messages for suppressed data dependent errors.

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #151023


 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv

laithsakka committed:
supress logging

[ghstack-poisoned]

laithsakka committed:
Update on "supress logging"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "supress logging"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "Do no generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "Do no generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "Do no generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

bobrenjc93 commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -1195,30 +1196,39 @@ def guard_or_false(a: BoolLikeType) -> bool:
+            return statically_known_true(a)
+        else:
+            try:
+                return bool(guard_bool(a))
```
gurad_bool already returns a bool?

bobrenjc93 replied:
```suggestion
                return guard_bool(a)
```

bobrenjc93 commented on torch/fx/experimental/symbolic_shapes.py:
```diff
@@ -1195,30 +1196,39 @@ def guard_or_false(a: BoolLikeType) -> bool:
             return True
+        else:
+            try:
+                return bool(guard_bool(a))
```
ditto

laithsakka committed:
Update on "Do not generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "Do not generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge

laithsakka committed:
Update on "Do not generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "Do not generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka committed:
Update on "Do not generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
@pytorch revert

laithsakka commented:
@pytorchbot revert

laithsakka commented:
@pytorchbot revert -m breaking other PRs -c weird

laithsakka commented:
@pytorchbot revert -m breaking other PRs -c weird

laithsakka commented:
@pytorchbot revert -m "breaking other PRs"  -c weird

laithsakka committed:
Update on "Do not generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
rebase

laithsakka committed:
Update on "Do not generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "Do not generate long log messaged for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "Do not generate long log messages for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka committed:
Update on "Do not generate long log messages for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge

malfet commented:
@pytorchbot merge -f "GH is broken"

laithsakka committed:
Update on "Do not generate long log messages for suppressed data dependent errors."



 TORCH_LOGS="all" python test/test_dynamic_shapes.py -k test_guard_or_true 
 
 before: 
<img width="1065" alt="Screenshot 2025-04-10 at 9 55 27 AM" src="https://github.com/user-attachments/assets/3ee20de0-2902-4eb1-8ab0-80f1b974fb78" />

after:
<img width="1124" alt="Screenshot 2025-04-10 at 9 54 35 AM" src="https://github.com/user-attachments/assets/4e7e1f0c-856c-417f-8763-bfe183e2450d" />


Note: we actually do not expect to see a log at all, this is an orthogonal issue in recording where it logs each error seen 
even when recording is not enabled? I will follow up with PR for that. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge -f "GH is broken" -i

laithsakka commented:
@pytorchbot merge -i
